runtime/scripts/optwin.lua

```diff
@@ -0,0 +1,661 @@
+    { 'termpastefilter', N_ 'characters removed when pasting into terminal window' },
+    { 'scrollback', N_ 'number of lines kept beyond the visible screen in terminal buffer' },
+  },
+}
+
+local lines = {
+  N_ '" Each "set" line shows the current value of an option (on the left).',
+  N_ '" Hit <Enter> on a "set" line to execute it.',
+  N_ '"            A boolean option will be toggled.',
+  N_ '"            For other options you can edit the value before hitting <Enter>.',
+  N_ '" Hit <Enter> on a help line to open a help window on this option.',
+  N_ '" Hit <Enter> on an index line to jump there.',
+  N_ '" Hit <Space> on a "set" line to refresh it.',
+  '',
+}
+for header_number, options in ipairs(options_list) do
+  table.insert(lines, ('%2d %s'):format(header_number, options.header))
+end
+table.insert(lines, '')
+
+local local_to_window = N_ '(local to window)'
+local local_to_buffer = N_ '(local to buffer)'
+local global_or_local_to_buffer = N_ '(global or local to buffer)'
+local global_or_local_to_window = N_ '(global or local to window)'
+
+for header_number, options in ipairs(options_list) do
+  table.insert(lines, '')
+  table.insert(lines, ('%2d %s'):format(header_number, options.header))
+  table.insert(lines, '')
+  for _, opt_desc_and_name in ipairs(options) do
+    local name, desc = unpack(opt_desc_and_name)
+    if #name >= 15 then
+      vim.list_extend(lines, vim.split(('%s\n\t%s'):format(name, desc:gsub('\n', '\n\t')), '\n'))
+    else
+      vim.list_extend(lines, vim.split(('%s\t%s'):format(name, desc:gsub('\n', '\n\t')), '\n'))
+    end
+    local info = vim.api.nvim_get_option_info2(name, {})
+
+    if info.scope == 'buf' and info.global_local then
+      table.insert(lines, '\t' .. global_or_local_to_buffer)
+    elseif info.scope == 'win' and info.global_local then
+      table.insert(lines, '\t' .. global_or_local_to_window)
+    elseif info.scope == 'buf' then
+      table.insert(lines, '\t' .. local_to_buffer)
+    elseif info.scope == 'win' then
+      table.insert(lines, '\t' .. local_to_window)
+    end
+
+    local shortname = info.shortname or name
+    if shortname == '' then
+      shortname = name
+    end
+    if info.type == 'boolean' then
+      if vim.o[name] then
+        table.insert(lines, (' \tset %s\tno%s'):format(shortname, shortname))
+      else
+        table.insert(lines, (' \tset no%s\t%s'):format(shortname, shortname))
+      end
+    else
+      local value = vim.o[name] --[[@as string]]
+      value = vim.fn.escape(value, ' \t\\"|')
+      table.insert(lines, (' \tset %s=%s'):format(shortname, value))
+    end
+  end
+end
+
+---@param line string
+---@return 'set'|'help'|'header'|'other'
+local function line_get_type(line)
+  if line:find('^ \tset ') then
+    return 'set'
+  elseif line:find('^%s-[0-9]') then
+    return 'header'
+  elseif line:find('^[\ta-z]') then
+    return 'help'
+  end
+  return 'other'
+end
+
+---@return integer|false
+local function find_alt_window()
```

Comment:
Not sure exactly what this does (a brief docstring would help) but is it something we should add to `runtime/lua/vim/_core/util.lua` ?